src/test/obj_alloc/obj_alloc.c

```diff
@@ -0,0 +1,147 @@
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * obj_alloc.c -- unit test for pmemobj_alloc and pmemobj_zalloc
+ */
+
+#include "unittest.h"
+#include "heap.h"
+#include <limits.h>
+
+POBJ_LAYOUT_BEGIN(alloc);
+POBJ_LAYOUT_ROOT(alloc, struct root);
+POBJ_LAYOUT_TOID(alloc, struct object);
+POBJ_LAYOUT_END(alloc);
+
+struct object {
+	size_t value;
+	char data[];
+};
+
+struct root {
+	TOID(struct object) obj;
+	char data[CHUNKSIZE - sizeof(TOID(struct object))];
+};
+
+static uint64_t
+check_int(const char *size_str)
+{
+	uint64_t ret;
+
+	switch (*size_str) {
+	case 'S':
+		ret = SIZE_MAX;
+		break;
+	case 'B':
+		ret = SIZE_MAX - 1;
+		break;
+	case 'O':
+		ret = sizeof(struct object);
+		break;
+	default:
+		ret = (uint64_t)STRTOLL(size_str, NULL, 10);
+	}
+	return ret;
+}
+
+int
+main(int argc, char *argv[])
+{
+	START(argc, argv, "obj_alloc");
+
+	const char *path;
+	size_t size;
+	uint64_t type_num;
+	int is_oid_null;
+	uint64_t flags;
+	int expected_return_code;
+	int expected_errno;
+	int ret;
+
+	if (argc < 8)
+		UT_FATAL("usage: %s path size type_num is_oid_null flags "
+		    "expected_return_code expected_errno ...", argv[0]);
+
+	PMEMobjpool *pop = NULL;
+	PMEMoid *oidp;
+
+	path = argv[1];
+	if ((pop = pmemobj_create(path, POBJ_LAYOUT_NAME(basic),
+		0, S_IWUSR | S_IRUSR)) == NULL)
+		UT_FATAL("!pmemobj_create: %s", path);
+
+	for (int i = 1; i + 6 < argc; i += 7) {
+		size = (size_t)check_int(argv[i + 1]);
+		type_num = check_int(argv[i + 2]);
+		is_oid_null = ATOI(argv[i + 3]);
+		flags = (uint64_t)STRTOLL(argv[i + 4], NULL, 10);
```

Comment:
with STRTOULL/ATOULL (the later one is preferred) you should not need to cast

Reply:
Done.